samples/subsys/fs/littlefs_blk/src/main.c

```diff
@@ -0,0 +1,272 @@
+		LOG_PRINTK("%02x ", p[i]);
+	}
+
+	LOG_PRINTK("\n");
+}
+
+
+static int littlefs_binary_file_adj(char *fname)
+{
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int rc;
+
+	/*
+	 * Uncomment below line to force re-creation of the test pattern
+	 * file on the littlefs FS.
+	 */
+	/* fs_unlink(fname); */
+	fs_file_t_init(&file);
+
+	rc = fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);
+	if (rc < 0) {
+		LOG_ERR("FAIL: open %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_stat(fname, &dirent);
+	if (rc < 0) {
+		LOG_ERR("FAIL: stat %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	/* Check if the file exists - if not just write the pattern */
+	if (rc == 0 && dirent.type == FS_DIR_ENTRY_FILE && dirent.size == 0) {
+		LOG_INF("Test file: %s not found, create one!",
+			log_strdup(fname));
+		init_pattern(file_test_pattern, sizeof(file_test_pattern));
+	} else {
+		rc = fs_read(&file, file_test_pattern,
+			     sizeof(file_test_pattern));
+		if (rc < 0) {
+			LOG_ERR("FAIL: read %s: [rd:%d]",
+				log_strdup(fname), rc);
+			return rc;
+		}
+		incr_pattern(file_test_pattern, sizeof(file_test_pattern), 0x1);
+	}
+
+	LOG_PRINTK("------ FILE: %s ------\n", fname);
+	print_pattern(file_test_pattern, sizeof(file_test_pattern));
+
+	rc = fs_seek(&file, 0, FS_SEEK_SET);
+	if (rc < 0) {
+		LOG_ERR("FAIL: seek %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_write(&file, file_test_pattern, sizeof(file_test_pattern));
+	if (rc < 0) {
+		LOG_ERR("FAIL: write %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_close(&file);
+	if (rc < 0) {
+		LOG_ERR("FAIL: close %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	return 0;
+}
+
+static int littlefs_increase_infile_value(char *fname)
+{
+	uint8_t boot_count = 0;
+	struct fs_file_t file;
+	int rc;
+
+	fs_file_t_init(&file);
+	rc = fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);
+	if (rc < 0) {
+		LOG_ERR("FAIL: open %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_read(&file, &boot_count, sizeof(boot_count));
+	if (rc < 0) {
+		LOG_ERR("FAIL: read %s: [rd:%d]", log_strdup(fname), rc);
+		return rc;
+	}
+	LOG_PRINTK("%s read count:%u (bytes: %d)\n", fname, boot_count, rc);
+
+	rc = fs_seek(&file, 0, FS_SEEK_SET);
+	if (rc < 0) {
+		LOG_ERR("FAIL: seek %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	boot_count += 1;
+	rc = fs_write(&file, &boot_count, sizeof(boot_count));
+	if (rc < 0) {
+		LOG_ERR("FAIL: write %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	LOG_PRINTK("%s write new boot count %u: [wr:%d]\n", fname,
+		   boot_count, rc);
+
+	rc = fs_close(&file);
+	if (rc < 0) {
+		LOG_ERR("FAIL: close %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	return 0;
+}
+
+/* mounting info */
+static struct fs_mount_t mp = {
+	.type = FS_LITTLEFS,
+	.fs_data = &lfsfs,
+	.flags = FS_MOUNT_FLAG_USE_DISK_ACCESS,
+};
+
+void main(void)
+{
+	static const char *disk_mount_pt = "/SDMMC:";
+	static const char *disk_pdrv = "SDMMC";
```

Comment:
This is a problem, it can be `CONFIG_SDMMC_VOLUME_NAME ` for now, but still not a good solution.

Reply:
not addressed